### ISSUE TITLE
PackageManager ...Flags classes introduced in API33 causes IncompatibleClassChangeError at robolectric tests

### ISSUE THREAD

**Issue by carstenhag** [robolectric/robolectric#7577]
### Description

API 33 introduced `PackageManager.ApplicationInfoFlags`, see https://developer.android.com/reference/android/content/pm/PackageManager.ApplicationInfoFlags. Using this class causes robolectric tests to fail.
Same thing applies to ResolveInfoFlags and probably also the other flag classes introduced. There are 4 in total: PackageInfoFlags, ApplicationInfoFlags, ComponentInfoFlags, ResolveInfoFlags.

```
Method 'android.content.pm.PackageManager$ApplicationInfoFlags android.content.pm.PackageManager$ApplicationInfoFlags.of(long)' must be InterfaceMethodref constant
java.lang.IncompatibleClassChangeError: Method 'android.content.pm.PackageManager$ApplicationInfoFlags android.content.pm.PackageManager$ApplicationInfoFlags.of(long)' must be InterfaceMethodref constant
	at com.bmw.chargenow.utils.FileLoggingTreeKt.appName(FileLoggingTree.kt:94)
	at com.bmw.chargenow.utils.FileLoggingTree$Companion.fileName(FileLoggingTree.kt:69)
	at com.bmw.chargenow.utils.FileLoggingTree$Companion.generateFile(FileLoggingTree.kt:80)
	at com.bmw.chargenow.App.configureLogger(App.kt:152)
	at com.bmw.chargenow.App.onCreate(App.kt:74)
```

### Steps to Reproduce
1. Use non-deprecated `PackageManager.getApplicationInfo` method
2. Robolectric tests now always result in exceptions

### Robolectric & Android Version
Robolectric 4.8.2, API 33 compileSdkVersion, Kotlin 1.7.10, jvmTarget 11

### Link to a public git repo demonstrating the problem:

```
New code:
        val applicationInfo = packageManager.getApplicationInfo(
            getApplicationInfo().packageName,
            PackageManager.ApplicationInfoFlags.of(0)
        )

Old code:
        val applicationInfo = packageManager.getApplicationInfo(getApplicationInfo().packageName, 0)
```

**Comment by hoisie** [comment]
Currently Robolectric supports a max API of 32 (S_V2). I am actively working on SDK 33 support. Until then you will not be able to use new SDK 33 APIs in Robolectric.

**Comment by carstenhag** [comment]
Oh sorry, I had not seen that in the readme :) No worries then.

**Comment by hoisie** [comment]
@carstenhag 4.9-alpha-1 was just released with SDK 33 support. Can you give that a try and let us know if there are any issues?

**Comment by carstenhag** [comment]
Sure. Still seems to be broken:

```

Method 'android.content.pm.PackageManager$ApplicationInfoFlags android.content.pm.PackageManager$ApplicationInfoFlags.of(long)' must be InterfaceMethodref constant
java.lang.IncompatibleClassChangeError: Method 'android.content.pm.PackageManager$ApplicationInfoFlags android.content.pm.PackageManager$ApplicationInfoFlags.of(long)' must be InterfaceMethodref constant
	at com.bmw.chargenow.utils.FileLoggingTreeKt.appName(FileLoggingTree.kt:92)
	at com.bmw.chargenow.utils.FileLoggingTree$Companion.fileName(FileLoggingTree.kt:69)
	at com.bmw.chargenow.utils.FileLoggingTree$Companion.generateFile(FileLoggingTree.kt:80)
	at com.bmw.chargenow.App.configureLogger(App.kt:152)
	at com.bmw.chargenow.App.onCreate(App.kt:74)
	at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1211)
	at org.robolectric.android.internal.RoboMonitoringInstrumentation.callApplicationOnCreate(RoboMonitoringInstrumentation.java:127)
	at org.robolectric.android.internal.AndroidTestEnvironment.lambda$installAndCreateApplication$2(AndroidTestEnvironment.java:369)
```

**Comment by utzcoz** [comment]
@carstenhag Could you provide a minimal reproduce app for this issue? And which JDK version do you use when encounter this problem?

**Comment by hoisie** [comment]
Is this a multi-sdk test? Are there tests running in multiple ask versions?

**Comment by hoisie** [comment]
I can sort of repro it, but only if I willingly use an SDK 33 method in SDK < 33:

```java
  @Test
  @Config(sdk = 32)
  public void invalid() throws Exception {
    Context context = ApplicationProvider.getApplicationContext();
    // causes IncompatibleClassChangeError trying to use ApplicationInfoFlags.of in SDK 32
    ApplicationInfo info =
        packageManager.getApplicationInfo(context.getPackageName(), ApplicationInfoFlags.of(0));
  }

  @Test
  @Config(sdk = 32)
  public void valid() throws Exception {
    Context context = ApplicationProvider.getApplicationContext();
    // works file
    ApplicationInfo info =
        (Build.VERSION.SDK_INT >= VERSION_CODES.TIRAMISU)
            ? packageManager.getApplicationInfo(
                context.getPackageName(), ApplicationInfoFlags.of(0))
            : packageManager.getApplicationInfo(context.getPackageName(), 0);
  }
```

@carstenhag can you verify you are doing an `SDK >= TIRAMISU` check when ` ApplicationInfoFlags.of` is used?

**Comment by carstenhag** [comment]
Some days ago I tried to setup a repro project, but it didn't work. Now with your help I figured out why :D 

So yeah, we were using `@Config(sdk = [Build.VERSION_CODES.P])`, probably it was needed 2 years ago to avoid some bug or something.

Also I just realized that the method is only available in Tiramisu, but Google decided not to add annotations (RequiredApi) because who needs linter warnings, let's just crash the app on runtime. Anyhow, the issue is solved, thanks!
